src/otp-input/otp-input.tsx

```diff
@@ -64,6 +65,12 @@ export const OtpInput = ({
         onChangeRef.current = onChange;
     }, [onChange]);
 
+    useEffect(() => {
+        if (value) {
```

Comment:
Here we might want to check if the length of array matches the `numOfInput` before assigning.